src/runtime-helpers.js

```diff
@@ -281,35 +281,91 @@ function returnAnnotations (action) {
-  if (!pathOnly) {
-    objectAPI.action = firstProp(packages[key]['apis'][api][objectAPI.basepath][objectAPI.relpath])
-    objectAPI.operation = packages[key]['apis'][api][objectAPI.basepath][objectAPI.relpath][objectAPI.action].method
-    objectAPI.responsetype = packages[key]['apis'][api][objectAPI.basepath][objectAPI.relpath][objectAPI.action].response || 'json' // binding the default parameter
-    if (ruleAction.includes(objectAPI.action)) {
-      if (packages[key]['actions'][objectAPI.action]['web'] || packages[key]['actions'][objectAPI.action]['web-export']) {
-        objectAPI.action = `${key}/${objectAPI.action}`
-      } else {
-        throw new Error('Action provided in api is not a web action')
-      }
-    } else if (arrSequence.includes(objectAPI.action)) {
-      if (packages[key]['sequences'][objectAPI.action]['web'] || packages[key]['sequences'][objectAPI.action]['web-export']) {
-        objectAPI.action = `${key}/${objectAPI.action}`
-      } else {
-        throw new Error('Sequence provided in api is not a web action')
-      }
-    } else {
-      throw new Error('Action provided in the api not present in the package')
-    }
+/**
+ * Creates an array of route definitions from the given manifest-based package.
+ * See https://github.com/apache/openwhisk-wskdeploy/blob/master/parsers/manifest_parser.go#L1187
+ *
+ * @param pkg The package definition from the manifest.
+ * @param pkgName The name of the package.
+ * @param apiName The name of the HTTP API definition from the manifest.
+ * @param allowedActions List of action names allowed to be used in routes.
+ * @param allowedSequences List of sequence names allowed to be used in routes.
+ * @param pathOnly Skip action, method and response type in route definitions.
+ * @return {{
+ * action: String,
+ * operation: String,
+ * responsetype: String
+ * basepath: String,
+ * relpath: String,
+ * name: String
+ * }[]}
+ */
+function createApiRoutes (pkg, pkgName, apiName, allowedActions, allowedSequences, pathOnly) {
+  const actions = pkg.actions
+  const sequences = pkg.sequences
+  const rawApi = pkg.apis[apiName]
+
+  if (!rawApi) {
+    throw new Error('Arguments to create API not provided')
   }
 
-  objectAPI.relpath = '/' + objectAPI.relpath
-  objectAPI.basepath = '/' + objectAPI.basepath
-  return objectAPI
+  const routes = []
+
+  Object.keys(rawApi).forEach((basePathName) => {
+    const basePath = rawApi[basePathName]
+
+    Object.keys(basePath).forEach((resourceName) => {
+      const resource = basePath[resourceName]
+
+      Object.keys(resource).forEach((actionName) => {
+        const route = {
+          name: apiName,
+          basepath: `/${basePathName}`,
+          relpath: `/${resourceName}`
+        }
+
+        // only name/path based information is requested
+        // return basic route here and return
+        if (pathOnly) {
+          routes.push(route)
+          return routes
```

Comment:
This is going to return only one `route`. What happens if `manifest.yml` has multiple `basepath/relpath` combinations?

Reply:
@ekremney will fix

Reply:
@ekremney return acts like a "continue" in a forEach loop, as each iteration is a funtion. The routes is unneccessary though.